server/rest/src/main/java/org/infinispan/rest/resources/CounterResource.java

```diff
@@ -43,33 +45,23 @@ public Invocations getInvocations() {
             .create();
    }
 
-   private NettyRestResponse getCounter(RestRequest request) throws RestResponseException {
-      try {
-         String counterName = request.variables().get("counterName");
-         String accept = request.getAcceptHeader();
-         MediaType contentType = accept == null ? MediaType.TEXT_PLAIN : negotiateMediaType(accept);
-         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
-         CounterConfiguration configuration = counterManager.getConfiguration(counterName);
-         if (configuration == null) {
-            return responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found").build();
-         } else {
-            responseBuilder.contentType(contentType);
-            switch (configuration.type()) {
-               case WEAK:
-                  long value = counterManager.getWeakCounter(counterName).getValue();
-                  responseBuilder.entity(Long.toString(value));
-                  break;
-               case BOUNDED_STRONG:
-               case UNBOUNDED_STRONG:
-                  value = counterManager.getStrongCounter(counterName).getValue().get(1, TimeUnit.MINUTES); // FIXME: make non-blocking in ISPN-10210
-                  responseBuilder.entity(Long.toString(value));
-                  break;
-            }
-         }
-         return responseBuilder.header(HttpHeaderNames.CACHE_CONTROL.toString(), CacheControl.noCache()).build();
-      } catch (Exception e) {
-         throw new RestResponseException(e);
+   private CompletionStage<RestResponse> getCounter(RestRequest request) throws RestResponseException {
+      String counterName = request.variables().get("counterName");
+      String accept = request.getAcceptHeader();
+      MediaType contentType = accept == null ? MediaType.TEXT_PLAIN : negotiateMediaType(accept);
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      CounterConfiguration configuration = counterManager.getConfiguration(counterName);
+      if (configuration == null) {
+         return CompletableFuture.completedFuture(responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found").build());
```

Comment:
Please move `responseBuilder.status(HttpResponseStatus.NOT_FOUND.code()).entity("Counter not found")` to a separate statement here too

Reply:
I have another PR with lots of change to ```CounterResource``` where this was done

Reply:
Ok, then I'm ready to integrate unless @wburns has any other objections.

Reply:
Nothing from me.